File: AI_A5_Vipul_2018118/AI_A5.py
import nltk
from nltk.corpus import stopwords 
from nltk.stem import WordNetLemmatizer
import string
import re
import inflect
nltk.download('wordnet')
nltk.download('stopwords')
from nltk.tokenize import word_tokenize   
import pickle
from num2words import num2words
lem=WordNetLemmatizer()

def user_answers(answer):
    temp=[]
    input1=input(answer)
    input1=input1.lower()
    input1=input1.translate(str.maketrans(" "," ",string.punctuation))
    final_tokens=[]
    input1= input1.split(" ")
    for tokens in input1:
        if(tokens.isdecimal()==True):
            tokens=num2words(tokens)
        word=lem.lemmatize(tokens)
        final_tokens.append(word)
    return final_tokens
file1 = open("E:/sem 7/AI/AI_A5_Vipul_2018118/facts.txt", 'w')


ans=user_answers("What kind of work you like? \n 1. Physical \n 2. Logical \n" )

if "logical" in ans:
    file1.write("physical_or_mental(mental).\n")
    ans1=user_answers("Having high aptitute skills? \n 1. Yes \n 2. No \n" ) 
    if "yes" in ans1:
        file1.write("aptitute(yes).\n")
        ans2=user_answers("Do you like technology or non-tech things? \n 1. Technical \n 2. Non-tech \n" ) 
        if "technical" in ans2:
            file1.write("tech_or_nontech(tech). \n")
            ans2=user_answers("Which exam you cleared? \n 1. CAT \n 2. GATE \n 3. GRE \n 4. None \n" )
            if "cat" in ans2:
                file1.write("given_exam(cat). \n")
            elif "gate" in ans2:
                file1.write("given_exam(gate). \n")
            elif "gre" in ans2:
                file1.write("given_exam(gre). \n")
            elif "none" in ans2:
                file1.write("given_exam(none). \n")
                ans3=user_answers("Do you like programming? \n 1. Yes \n 2. No \n")
                if "yes" in ans3:
                    file1.write("coding(yes). \n")
                else:
                    file1.write("coding(no). \n")
        else:
            file1.write("tech_or_nontech(nontech). \n")
            ans41=user_answers("do you have interest in doing business? \n 1. Yes 2. No \n")
            if "yes" in ans41:
                file1.write("business_oriented(yes). \n")
            else:
                file1.write("business_oriented(no). \n")
                ans4= user_answers(" Do you like to travel? \n 1. Traveller \n 2. Non-Traveller \n")
                if "traveller" in ans4:
                    file1.write("traveller_nontraveller(traveller). \n")
                    ans5= user_answers(" Do you like creativity and innovation? \n 1. Yes \n 2. No \n")
                    if "yes" in ans5:
                        file1.write("creative_and_innovative(yes). \n")
                else:
                    file1.write("traveller_nontraveller(nontraveller). \n")
                    ans5= user_answers(" Are you a person of creative and innovative ideas? \n 1. Yes \n 2. No \n")
                    if "yes" in ans5:
                        file1.write("creative_and_innovative(yes). \n")
    else:
        file1.write("aptitute(no). \n")
        ans6=user_answers("Do you like technology or non-tech things? \n 1. Technical \n 2. Non-tech \n" ) 
        if "technical" in ans6:
            file1.write("tech_or_nontech(tech). \n")
            ans7= user_answers(" Do you like to travel? \n 1. Traveller \n 2. Non-Traveller ")
            if "traveller" in ans7:
                    file1.write("traveller_nontraveller(traveller). \n")
                    ans8= user_answers(" Are you a person of creative and innovative ideas? \n 1. Yes \n 2. No \n")
                    if "yes" in ans8:
                        file1.write("creative_and_innovative(yes). \n")
            else:
                    file1.write("traveller_nontraveller(nontraveller). \n")
                    ans9= user_answers(" Are you a person of creative and innovative ideas? \n 1. Yes \n 2. No \n")
                    if "yes" in ans9:
                        file1.write("creative_and_innovative(yes). \n")
        else:
            file1.write("tech_or_nontech(nontech). \n")
            ans10= user_answers(" Do you like to travel? \n 1. Traveller \n 2. Non-Traveller ")
            if "traveller" in ans10:
                    file1.write("traveller_nontraveller(traveller). \n")
                    ans11= user_answers(" Are you a person of creative and innovative ideas? \n 1. Yes \n 2. No \n")
                    if "yes" in ans11:
                        file1.write("creative_and_innovative(yes). \n")
            else:
                    file1.write("traveller_nontraveller(nontraveller). \n")
                    ans12= user_answers(" Are you a person of creative and innovative ideas? \n 1. Yes \n 2. No \n")
                    if "yes" in ans12:
                        file1.write("creative_and_innovative(yes). \n")

elif "physical" in ans:
    file1.write("physical_or_mental(physical).\n")
    ans1=user_answers("Do you have leadership skills? \n 1. Yes \n 2. No \n" )
    if "yes" in ans1:
        file1.write("leadership(yes).\n")
        ans2=user_answers("Do you like taking risk? \n 1. Yes \n 2. No \n" )
        if "yes" in ans2:
            file1.write("risk_taker(yes). \n")
            ans3=user_answers("Having high aptitute skills? \n 1. Yes \n 2. No \n" )
            if "yes" in ans3:
                file1.write("aptitute(yes).\n")
        
                
        
    
print("Your facts are successfully saved in facts.txt file. \n Now you can use prolog for advicing career")

file1.close()

File: AI_A5_Vipul_2018118/Ai_A5.pl

start :-
  ansi_format([bold,fg(green)], 'Career Advisory System. ~w', ['']),nl,
  ansi_format([bold], 'Welcome !! We will try to advice best career option for you. ~w', ['']),nl,
  write('We will give you different options.'), nl,
  reset_values,
  readFromFile,
  advice_career(Option).


readFromFile:-
  setup_call_cleanup(
    open('E:/sem 7/AI/AI_A5_Vipul_2018118/facts.txt',read,Input),
    (
      repeat,
      read_term(Input, X, []),
      read_from_file(X),asserta(X), !
    ),
    close(Input)
  ).


advice_career(Option) :-career(Option), !.

:- dynamic(result/2).


reset_values :-retract(result(_, _)),fail.
reset_values.



career(ms) :-
  physical_or_mental(mental),
  aptitute(yes),
  tech_or_nontech(tech),
  given_exam(gre),
  write('You look  to be interested in research.'),nl,
  write('So, you can go for higher studies'),nl,
  ansi_format([bold,fg(magenta)], ' Suggestion: Masters in Science ~w', ['']).


career(mtech) :-
  physical_or_mental(mental),
  aptitute(yes),
  tech_or_nontech(tech),
  given_exam(gate),
  write(' '),
  write('You seems to be interested in research.'),nl,
  write('You have cleared gate cutoff and you are interested in technology.'),nl,
  write('So, you can go for higher studies'),nl,
  ansi_format([bold,fg(magenta)], ' Suggestion: Masters in Technology. ~w', ['']).


career(mba) :-
  physical_or_mental(mental),
  aptitute(yes),
  tech_or_nontech(tech), 
  given_exam(cat),
  write('You seems to be interested in business.'),nl,
  write('You have cleared CAT cutoff and you have good management skills.'),nl,
  write('You can go for higher studies'),nl,
  ansi_format([bold,fg(magenta)], ' Suggestion: Masters in Business Administration. ~w', ['']).

  career(job) :-
  physical_or_mental(mental),
  aptitute(yes),
  tech_or_nontech(tech), 
  given_exam(none),
  coding(yes),
  write('You seems to be interested in coding .'),nl,
  write('So, you can go for an IT Job'),nl,
  write('We would suggest you following job options'),nl,
  ansi_format([bold,fg(magenta)], ' 1. Software developer. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 2. Data Anayst. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 3. Technical researcher. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 4. Cyber Security Analyst. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 5. System Designer. ~w', ['']).

  career(others) :-
  physical_or_mental(mental),
  aptitute(no),
  tech_or_nontech(tech),  
   business_oriented(no),
  traveller_nontraveller(traveller),
  creative_and_innovative(yes),
  write('You look like a creative and innovative person with technical knowledge.'),nl,
  write('You  like to travel a lot. '),nl,
  write('We would suggest you following job options'),nl,
  ansi_format([bold,fg(magenta)], ' 1. Event Planner. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 2. Journalism. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 3. Blogger. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 4. Photographer if you are interested in photography.. ~w', ['']).


  career(others) :-
  physical_or_mental(mental),
  aptitute(no),
  tech_or_nontech(nontech),  
  traveller_nontraveller(nontraveller),
  creative_and_innovative(yes),
  write('You seems like a creative and innovative person.'),nl,
  write('But you do not like travelling and wants to do something by staying at a place only'),nl,
  write('We would suggest you following job options'),nl,
  ansi_format([bold,fg(magenta)], ' 2. Interior Designer. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 2. Artist. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 3. Fashion Designer.. ~w', ['']).


career(job) :-
  physical_or_mental(mental),
  aptitute(yes),
  tech_or_nontech(tech),
  given_exam(none),
  coding(no),
  write('You do not like coding.'),nl,
  write('We would suggest you following job options'),nl,
  ansi_format([bold,fg(magenta)], ' 1. Public Sector. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 2. Teaching profile. ~w', ['']).

career(army) :-
  physical_or_mental(physical),
  aptitute(yes),
  leadership(yes),
  risk_taker(yes),
  write('You look like  to be physically fit.'),nl,
  write('You are a risk taker and can adapt to any situation.'),nl,
  write('We would suggest you following job options'),nl,
  ansi_format([bold,fg(magenta)], ' 1. Join India Army Force. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 2. Join India Air Force. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 3. Join India Navy Force. ~w', ['']).

career(business) :-
  physical_or_mental(mental),
  aptitute(yes),
  tech_or_nontech(nontech),   
  business_oriented(yes),
  write('You are a non-technological person but you are good in aptitute and you are a risk taker.'),nl,
  write('You are interested in business and you have good management skills and well as you are a good leadership.'),nl,
  write('We would suggest you following job options'),nl,
  ansi_format([bold,fg(magenta)], ' 1. You can start your own business. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 2. Be an entrepreneaur. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 3. Join any business related company. ~w', ['']).





career(others) :-
  physical_or_mental(mental),
  aptitute(yes),
  tech_or_nontech(nontech),  
  business_oriented(no),
  traveller_nontraveller(nontraveller),
  creative_and_innovative(yes),
  write('You seems like a creative and innovative person.'),nl,
  write('But you do not like travelling and wants to do something by staying at a place only'),nl,
  write('We would suggest you following job options'),nl,
  ansi_format([bold,fg(magenta)], ' 2. Interior Designer. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 2. Artist. ~w', ['']),nl,
  ansi_format([bold,fg(magenta)], ' 3. Fashion Designer.. ~w', ['']).







career(others):-
ansi_format([bold,fg(green)], 'Sorry, we are not able to predict you the career in future. we were not able to get your all details. ~w', ['']),nl,
write('You can take try a hand on new techs and explore your self.').

read_from_file(end_of_file):- !.
read_from_file(X):-
  asserta(X),!,
  fail.